### PR TITLE
[14.0][POC] generate mis report from account.tax.report

### PR DESCRIPTION
**Pull request by sbidoul**
This is a proof of concept that illustrates how to generate a MIS Report template from `account.tax.report` objects that are available in many localizations in Odoo CE.

This will probably not generate a ready to use report but this at least gives a solid basis to work on and finish manually by adding styles and tweaking the formulas if needed.

### CODE REVIEW

**Comment by rvalyi** [comment]
Hello @sbidoul, just to know, do you think the same could be backported to v12, or is there some big catch if we try that on v12? Thank you for this module!

**Comment by sbidoul** [comment]
> Hello @sbidoul, just to know, do you think the same could be backported to v12, or is there some big catch if we try that on v12? 

Do account.tax.report and account.tax.report.line exist in v12 ? I've not checked TBH.

> Thank you for this module!

You're welcome, thanks for saying thanks, it's always appreciated :) Although I'd not call this a module yet. There is no UI etc. I don't think I'll have much time to go further with this, but @vdewulf's team and other may pick this up and continue.

**Comment by sbidoul** [comment]
Ah, another caveat: this does not work with cash basis taxes.

**Comment by rvalyi** [comment]
I see, there is no such account.tax.report(.line) on v12 indeed... But anyway we may use it after v14.

**Comment by vdewulf** [comment]
@sbidoul thanks for this :) I'll let you know how it goes on our side!

File: mis_builder_tax_report/models/mis_builder_tax_report.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        "We don't grok this."
+                    ).format(name=report_line.name),
+                )
+            if report_line.tag_name:
+                # If the report line has tags, make an expression that gets the balance
+                # of move lines with these tags over the period.
+                kpi.expression = (
+                    f'-balp[][("tax_tag_ids.name", "=", "+{report_line.tag_name}")]'
+                    f'+balp[][("tax_tag_ids.name", "=", "-{report_line.tag_name}")]'
+                )
```
Comment (sbidoul) [inline mis_builder_tax_report/models/mis_builder_tax_report.py:77]: Note to self and interested parties. This formula is too simplistic for some localization. In v14 one must take into account the type of entry and possibly also the journal. In v15 this rule is computed into a tax_tag_invert field on account.move.line that also influences the sign to be applied in the report. I have not checked v13.

So this is to be used with caution and careful testing.

Comment (sbidoul) [reply mis_builder_tax_report/models/mis_builder_tax_report.py:77]: @vdewulf

**Comment by elvise** [comment]
@sbidoul good PR!

**Pull request closed without merging**